drivers/include/sx127x.h

```diff
@@ -0,0 +1,688 @@
+};
+
+/**
+ * @brief   LoRa error coding rate.
+ */
+enum {
+    SX127X_CR_4_5 = 1,                                          /**< coding rate 4/5 */
+    SX127X_CR_4_6,                                              /**< coding rate 4/6 */
+    SX127X_CR_4_7,                                              /**< coding rate 4/7 */
+    SX127X_CR_4_8                                               /**< coding rate 4/8 */
+};
+
+/**
+ * @brief   Radio driver internal state machine states definition.
+ */
+enum {
+    SX127X_RF_IDLE = 0,                                         /**< Idle state */
+    SX127X_RF_RX_RUNNING,                                       /**< Sending state */
+    SX127X_RF_TX_RUNNING,                                       /**< Receiving state */
+    SX127X_RF_CAD,                                              /**< Channel activity detection state */
+};
+
+/**
+ * @brief   Event types.
+ */
+enum {
+    SX127X_RX_DONE = 0,                                         /**< Receiving complete */
+    SX127X_TX_DONE,                                             /**< Sending complete*/
+    SX127X_RX_TIMEOUT,                                          /**< Receiving timeout */
+    SX127X_TX_TIMEOUT,                                          /**< Sending timeout */
+    SX127X_RX_ERROR_CRC,                                        /**< Receiving CRC error */
+    SX127X_FHSS_CHANGE_CHANNEL,                                 /**< Channel change */
+    SX127X_CAD_DONE,                                            /**< Channel activity detection complete */
+};
+
+/**
+ * @brief   LoRa configuration structure.
+ */
+typedef struct {
+    uint8_t power;                                              /**< Signal power */
+    uint8_t bandwidth;                                          /**< Signal bandwidth */
+    uint8_t datarate;                                           /**< Spreading factor rate, e.g datarate */
+    bool low_datarate_optimize;                                 /**< Optimize for low data rate */
+    uint8_t coderate;                                           /**< Error coding rate */
+    uint16_t preamble_len;                                      /**< Length of preamble header */
+    bool use_fix_len;                                           /**< Use fixed header len, (enable implicit header mode) */
+    uint8_t payload_len;                                        /**< Payload length */
+    bool crc_on;                                                /**< Enable payload CRC */
+    bool freq_hop_on;                                           /**< Enable frequency hopping */
+    uint8_t freq_hop_period;                                    /**< Frequency hop period */
+    bool iq_inverted;                                           /**< Set inverted IQ */
+    bool rx_continuous;                                         /**< Use continuous reception */
```

Comment:
> There are other bools in this struct that can all be summarized to a flag field, saving tons of bytes

I know, but I didn't want to get into optimizations first and have a clean and working driver implementation